doc/Beam_splitter.md

```diff
@@ -0,0 +1,73 @@
+
+*Note: this file is an early design document, but might be updated as the version goes up.*
+
+# Beam splitter #
+
+This article is primarily for reference, since we don't plan to make
+beam splitters separate entities in the simulator.
+
+## Photon Number States ##
+
+When you see  $$|0\rangle$$, $$|1\rangle$$, $$|2\rangle$$, etc. in a
+paper on quantum optics, be aware that it likely represents a photon
+number state, also called Fock state, I believe.  The 0, 1, 2
+represents the number of photons in a particular place and time
+(called a _spatial mode_).
+
+For photon number (or spatial mode) qubits, photon statistics at beam
+splitters are complicated.
+
+See:
+
+1. Weihs and Zeilinger lecture notes from Caltech QIP course
+1. Sec. 6.2 & 6.3 of Gerry & Knight
+1. Campos et al., PRA 40(3), 1371, 1989, esp. Eqs. 24, 42-48
+
+## Polarization States ##
+
+When you see $$|H\rangle$$ and $$|V\rangle$$ states, they refer to the
+polarization of a photon.  It's important to realize, though, that
+they represent the polarization of photon in a particular place and at
+a fixed time.  Weihs and Zeilinger refer to it as spin representing an
+_additional_ degree of freedom, and are very explicit about writing
+out the behavior of the BS in four bases: H and V at each of the two
+input ports.  This allows the unitary to be written properly, with the
+after state being a four-basis representation at the two output ports.
+
+Based on Eqs. 1.21 & 1.21 of W&Z, in the basis
+
+$$\left(\begin{matrix}
```

Comment:
These latex texts can't be recognized in Github markdown. You may have to use third-party tools to shot it correctly.